src/rpm/systemd-update-helper.c

```diff
@@ -0,0 +1,912 @@
+                                                       *user, bus_error_message(&error, r));
+                                        continue;
+                                }
+
+                                _cleanup_strv_free_ char **paths = NULL;
+                                r = sd_bus_message_read_strv(reply, &paths);
+                                if (r < 0)
+                                        return bus_log_parse_error(r);
+
+                                STRV_FOREACH(path, paths) {
+                                        r = bus_wait_for_jobs_add(w, *path);
+                                        if (r < 0)
+                                                return log_error_errno(r, "Failed to watch job %s: %m", *path);
+                                }
+
+                                r = bus_wait_for_jobs(w, BUS_WAIT_JOBS_LOG_ERROR, /* extra_args= */ NULL);
+                                if (r < 0)
+                                        return r;
+                        }
+                }
+        }
+
+        return 0;
+}
+
+static int verb_system_reload_restart(int argc, char **argv, void *userdata) {
+        return reload_restart(argc, argv, RUNTIME_SCOPE_SYSTEM);
+}
+
+static int verb_user_reload_restart(int argc, char **argv, void *userdata) {
+        return reload_restart(argc, argv, RUNTIME_SCOPE_GLOBAL);
+}
+
+static int run(int argc, char *argv[]) {
+        int r;
+
+        static const Verb verbs[] = {
+                { "install-system-units",      1, VERB_ANY, 0, verb_install_system_units      },
+                { "install-user-units",        1, VERB_ANY, 0, verb_install_user_units        },
+                { "remove-system-units",       1, VERB_ANY, 0, verb_remove_system_units       },
+                { "remove-user-units",         1, VERB_ANY, 0, verb_remove_user_units         },
+                { "mark-restart-system-units", 2, VERB_ANY, 0, verb_mark_restart_system_units },
+                { "mark-reload-system-units",  2, VERB_ANY, 0, verb_mark_reload_system_units  },
+                { "mark-restart-user-units",   2, VERB_ANY, 0, verb_mark_restart_user_units   },
+                { "mark-reload-user-units",    2, VERB_ANY, 0, verb_mark_reload_user_units    },
+                { "system-reload-restart",     1, 1,        0, verb_system_reload_restart     },
+                { "system-reload",             1, 1,        0, verb_system_reload_restart     },
+                { "system-restart",            1, 1,        0, verb_system_reload_restart     },
+                { "user-reload-restart",       1, 1,        0, verb_user_reload_restart       },
+                { "user-reload",               1, 1,        0, verb_user_reload_restart       },
+                { "user-restart",              1, 1,        0, verb_user_reload_restart       },
+                { "user-reexec",               1, 1,        0, verb_user_reload_restart       },
```

Comment:
This all sounds like a call for proper RuntimeScope option. Especially if this is going to be used across distros let's clean up the interface and not carry over the repetitive verbs

Reply:
There are existing external users of the helper: https://sourcegraph.com/search?q=context:global+repo:%5Esrc.fedoraproject.org/+systemd-update-helper&patternType=regexp&sm=0

I'm fine with making the documented interface something better, but let's keep compat by still having the old verbs as well.

Reply:
Sure, but you can do a trivial shim under rpm/ and move this under update-helper/?